hack/logcollector/cmd/template.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/edgelesssys/constellation/v2/hack/logcollector/internal"
+	"github.com/spf13/cobra"
+)
+
+type infoMap map[string]string
+
+var defaultInfoMap = infoMap{
+	"is-debug-cluster": "false",
+}
+
+func newTemplateCmd() *cobra.Command {
+	templateCmd := &cobra.Command{
+		Use:   "template",
+		Short: "Templates filebeat and logstash configurations and prepares them for deployment",
+		Long:  `Templates filebeat and logstash configurations and prepares them for deployment by placing them in the specified directory.`,
+		RunE:  runTemplate,
+	}
+
+	templateCmd.Flags().String("dir", "", "Directory to place the templated configurations in (required)")
+	must(templateCmd.MarkFlagRequired("dir"))
+	must(templateCmd.MarkFlagDirname("dir"))
+	templateCmd.Flags().String("username", "", "OpenSearch username (required)")
+	must(templateCmd.MarkFlagRequired("username"))
+	templateCmd.Flags().String("password", "", "OpenSearch password (required)")
+	must(templateCmd.MarkFlagRequired("password"))
+	templateCmd.Flags().Int("port", 5045, "Logstash port")
+	templateCmd.Flags().StringToString("info", nil, "Additional fields for the Logstash pipeline in the format --info key1=value1,key2=value2,...")
```

Comment:
Please validate that flags have valid keys defined in `fields.go`

Reply:
As its in debugd's internal package right now and thus can't be imported in `/hack`, should we maybe take the time and build a top-level internal package for this?

Reply:
Or move it to hack and import in debugd? Otherwise ok for to move the pkg.